.github/workflows/release-upload.yaml

```diff
@@ -94,29 +96,44 @@ jobs:
       timeout-minutes: 120
       run: |
         while true; do
-          # When supporting other architectures than Linux, this query should be adapted:
-          conclusion=$(gh api "repos/$GITHUB_REPOSITORY/commits/${{ env.TARGET_TAG }}/check-runs" --jq '.check_runs[] | select(.name | test("ci/hydra-build:.*-linux.required")) | .conclusion')
-          case "$conclusion" in
-            success)
-              echo "ci/hydra-build:required succeeded"
-              exit 0;;
-            failure)
-              echo "ci/hydra-build:required failed"
-              exit 1;;
-            *)
-              echo "ci/hydra-build:required pending. Waiting 30s..."
-              sleep 30;;
-          esac
+          conclusion=$(gh api "repos/$GITHUB_REPOSITORY/commits/${{ env.TARGET_TAG }}/check-runs" --jq '.check_runs[] | select(.name | test("ci/hydra-build:.*\\.required")) | .conclusion')
+          # Here we are being careful, because we query the status of multiple jobs (once per line)
+          # But the only thing we are sure is that "success" means a green job. There
+          # could be unknown statuses, which is why we may retry when unsure (see 'sleep') below.
+          echo "ci/hydra-build:.*\\.required returned status: $conclusion"
+          # conclusion is of the form (note the newlines, which matter because we use 'wc -l' below)
+          # success
+          # failure
+          # success
+          # Because we care of the newlines, quoting $conclusion with "" is especially important below!
+          # (see https://stackoverflow.com/questions/22101778/how-to-preserve-line-breaks-when-storing-command-output-to-a-variable)
+
+          # shellcheck disable=SC2126
```

Comment:
shellcheck wants us to use `grep -c` instead of `grep .. | wc -l`. The problem with `grep -c` is that it fails if the number of matches is 0. We don't want that, so we keep the usage of `wc -l`, but need to turn the warning off.